### ISSUE TITLE
[🐞]useSignal only triggers a re-render if its value is compared, but not when read

### ISSUE THREAD

**Issue by n8sabes** [BuilderIO/qwik#2336]
### Which component is affected?

Qwik Runtime

### Describe the bug

The attached stackblitz example switches themes by updating vanilla-extract variables. There is no reason to compare the signal value, only consume it. However, without a comparison a re-render never happens. Is a signal value comparison required to trigger a re-render, or is this a misunderstanding of how `useSignal` is used?

### Reproduction

https://stackblitz.com/edit/qwik-starter-qk2t3v?file=src%2Froutes%2Findex.tsx

### Steps to reproduce

**LINE 23 WORKS:** `<div className={theme.value || theme.value}>`

**LINE 24 DOES NOT WORK:** `<div className={theme.value}>`

[In StackBlitz, comment line 23 out, and uncomment line 24 to reproduce issue](https://stackblitz.com/edit/qwik-starter-qk2t3v?file=src%2Froutes%2Findex.tsx)

### System Info

```shell
System:
    OS: macOS 12.6
    CPU: (8) arm64 Apple M1
    Memory: 204.94 MB / 16.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 19.0.1 - ~/.nvm/versions/node/v19.0.1/bin/node
    npm: 8.19.2 - ~/.nvm/versions/node/v19.0.1/bin/npm
  Browsers:
    Chrome: 107.0.5304.121
    Firefox Developer Edition: 108.0
    Safari: 16.0
  npmPackages:
    @builder.io/qwik: ^0.14.1 => 0.14.1
    @builder.io/qwik-city: ^0.0.127 => 0.0.127
    @builder.io/qwik-react: ^0.1.1 => 0.1.1
    vite: 3.2.4 => 3.2.4
```

### Additional Information

[Discord Help DIscussion](https://discord.com/channels/842438759945601056/1047434791601057853/1047434791601057853)

**Comment by n8sabes** [comment]
***Shared by shiroinegai in Discord:***

"I think the issue Jeremy posted might be related to why it doesn't work but in the meantime, I found that changing it to this works:
```
export default component$(() => {
  const isDarkMode = useSignal(true);

  return (
    <>
      <div className={isDarkMode.value ? darkTheme : lightTheme}>
        <div
          className={container}
          onClick$={() => {
            isDarkMode.value = !isDarkMode.value;
          }}
        >
          Click to Switch Theme
          ...
        </div>
      </div>
    </>
  );
});
```
For some reason, moving the ternary up to where the class is injected works while doing so in the click handler doesn't 🤔"

**Comment by n8sabes** [comment]
Fixed in `0.15.0`

🎉
